### DISCUSSION TITLE
ATTACH for parquet on S3

### DISCUSSION THREAD

**Discussion by OneCyrus** [duckdb/duckdb#14208] [Ideas]
we have quite some raw data in the parquet format on S3. currently we always copy all the content from those parquet files to the duckdb file to process the data. but wouldn't it be cool if we could just attach the parquet file like a sqlite or postgresql? and only query the data from the remote storage we are actually using in the queries?

basically something like:
```
ATTACH 's3://mybucket/test.parquet' as test_table (READ_ONLY);
SELECT id from test_table;
```

instead of
```
COPY test_table FROM 's3://mybucket/test.parquet';
SELECT id from test_table;
```

sure we could only copy selected columns. but usually the needed columns are not known when copying the data and the analys querying the data should have potential access to all columns without re-importing the data.

**Comment by szarnyasg** [comment]
You can create a view on the Parquet file:
```
CREATE VIEW v AS FROM 's3://mybucket/test.parquet';
```
This will give access to the list of columns. [Partial reading is supported](https://duckdb.org/docs/extensions/httpfs/s3api#partial-reading), so DuckDB will push down column access to the Parquet files in S3.

**Reply by OneCyrus** [reply]
good to know :)

i guess there's no way around re-creating the view when the schema in the parquet file changes? has the attach function the same behavior. do i need to re-attach once the underlying database changed?
